### PR TITLE
Add legacy update test

### PR DESCRIPTION
**Pull request by gosusnp**
## What
* Add acceptance test for the Legacy state on update connection when adding a stream.

### CODE REVIEW

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java [hunk 1]
```diff
@@ -857,6 +867,168 @@ public void testSyncAfterUpgradeToPerStreamStateWithNoNewData(final TestInfo tes
...
+
+  private JobRead waitUntilTheNextJobIsStarted(final UUID connectionId) throws Exception {
+    final JobRead lastJob = getMostRecentSyncJobId(connectionId);
+    if (lastJob.getStatus() != JobStatus.SUCCEEDED) {
+      return lastJob;
+    }
+
+    JobRead mostRecentSyncJob = getMostRecentSyncJobId(connectionId);
+    while (mostRecentSyncJob.getId().equals(lastJob.getId())) {
+      Thread.sleep(Duration.ofSeconds(10).toMillis());
```
Comment (benmoriceau) [inline airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java:1026]: should we decrease that, 10 seconds seems big.

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java [hunk 2]
```diff
@@ -953,6 +1125,7 @@ public void testIncrementalSyncMultipleStreams() throws Exception {
   // This test is disabled because it takes a couple of minutes to run, as it is testing timeouts.
   // It should be re-enabled when the @SlowIntegrationTest can be applied to it.
   // See relevant issue: https://github.com/airbytehq/airbyte/issues/8397
+  @Test
```
Comment (benmoriceau) [inline airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java:1128]: Do you want to re-enable this test?

Comment (gosusnp) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java:1128]: No, the `@Disabled` is still below. I expected the `@Test` to have been removed as part as a not so great merge conflict resolve.

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java (hunk 1 above)
Comment (gosusnp) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java]: Lowered it to 1 second. In practice we are still looking at a fairly slow operation, I don't think it makes sense to lower it below a second.

**Review by benmoriceau — APPROVED** [review]

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java [hunk 3]
```diff
@@ -857,6 +867,168 @@ public void testSyncAfterUpgradeToPerStreamStateWithNoNewData(final TestInfo tes
...
+    // Some inspection for debug
+    destDb.query(ctx -> {
+      prettyPrintTables(ctx, outputPrefix + sourceTable1, outputPrefix + sourceTable2);
+      return null;
+    });
+    final ConnectionState initSyncState =
+        apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connection.getConnectionId()));
+    LOGGER.info("ConnectionState after the initial sync: " + initSyncState.toString());
+
+    testHarness.assertSourceAndDestinationDbInSync(false);
```
Comment (lmossman) [inline airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java:923]: nit: could you use the `WITH_SCD_TABLE` and `WITHOUT_SCD_TABLE` constants in the calls to this method, to improve the readability a bit?

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java [hunk 4]
```diff
@@ -857,6 +867,168 @@ public void testSyncAfterUpgradeToPerStreamStateWithNoNewData(final TestInfo tes
...
+                .operatorConfiguration(operation.getOperatorConfiguration())))
+        .syncCatalog(updatedCatalog)
+        .schedule(connection.getSchedule())
+        .sourceCatalogId(connection.getSourceCatalogId())
+        .status(connection.getStatus())
+        .resourceRequirements(connection.getResourceRequirements())
+        .withRefreshedCatalog(true);
+    final WebBackendConnectionRead connectionUpdateRead = webBackendApi.webBackendUpdateConnection(update);
+
+    LOGGER.info("Inspecting Destination DB after the update request, tables should be empty");
```
Comment (lmossman) [inline airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java:970]: This says that we expect the dest tables to be empty, but we don't actually programmatically enforce that here. Should we add some assertion that the tables are actually empty to codify that?

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java [hunk 5]
```diff
@@ -857,6 +867,168 @@ public void testSyncAfterUpgradeToPerStreamStateWithNoNewData(final TestInfo tes
...
+      return lastJob;
+    }
+
+    JobRead mostRecentSyncJob = getMostRecentSyncJobId(connectionId);
+    while (mostRecentSyncJob.getId().equals(lastJob.getId())) {
+      Thread.sleep(Duration.ofSeconds(1).toMillis());
+      mostRecentSyncJob = getMostRecentSyncJobId(connectionId);
+    }
+    return mostRecentSyncJob;
+  }
```
[the remark this answers is not in the document]
Comment (gosusnp) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java:1030]: https://github.com/airbytehq/airbyte/pull/14656

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java [hunk 6]
```diff
@@ -857,6 +867,168 @@ public void testSyncAfterUpgradeToPerStreamStateWithNoNewData(final TestInfo tes
     testHarness.assertSourceAndDestinationDbInSync(WITHOUT_SCD_TABLE);
   }
 
+  @Test
+  public void testResetAllWhenSchemaIsModified() throws Exception {
```
[the remark this answers is not in the document]
Comment (gosusnp) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java:871]: Good catch and yes, once we have enabled per-stream state everywhere, we should remove this test.
Until then, we should have a test to ensure we do not break this behavior, I am thinking of forcing the use of a non stream source, maybe mysql?

**Pull request merged**